# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/nsIThreadInternal.idl
# Commit: 36ef70762b74
# Full Hash: 36ef70762b74b3c6b8bd0f26c57ab4b54467f64b
# Author: Bill McCloskey <billm@mozilla.com>
# Date: 2017-08-17 23:17:14
# Regressor Bug: 1382922
# File Overlap Count: 3
# Description:
#   Bug 1382922 - Refactor event queue to allow multiple implementations (r=erahm)
#   
#   This patch refactors the nsThread event queue to clean it up and to make it easier to restructure. The fundamental concepts are as follows:
#   
#   Each nsThread will have a pointer to a refcounted SynchronizedEventQueue. A SynchronizedEQ takes care of doing the locking and condition variable work when posting and popping events. For the actual storage of events, it delegates to an AbstractEventQueue data structure. It keeps a UniquePtr to the AbstractEventQueue that it uses for storage.
# ==============================================================================

diff -r ca6618d0bc17 -r 36ef70762b74 xpcom/threads/nsIThreadInternal.idl
--- a/xpcom/threads/nsIThreadInternal.idl	Tue Aug 01 12:55:21 2017 -0400
+++ b/xpcom/threads/nsIThreadInternal.idl	Tue Jun 20 19:42:13 2017 -0700
@@ -40,28 +40,6 @@
    * observer will never be called again by the thread.
    */
   void removeObserver(in nsIThreadObserver observer);
-
-  /**
-   * This method causes any events currently enqueued on the thread to be
-   * suppressed until PopEventQueue is called, and any event dispatched to this
-   * thread's nsIEventTarget will queue as well. Calls to PushEventQueue may be
-   * nested and must each be paired with a call to PopEventQueue in order to
-   * restore the original state of the thread. The returned nsIEventTarget may
-   * be used to push events onto the nested queue. Dispatching will be disabled
-   * once the event queue is popped. The thread will only ever process pending
-   * events for the innermost event queue. Must only be called on the target
-   * thread.
-   */
-  [noscript] nsIEventTarget pushEventQueue();
-
-  /**
-   * Revert a call to PushEventQueue. When an event queue is popped, any events
-   * remaining in the queue are appended to the elder queue. This also causes
-   * the nsIEventTarget returned from PushEventQueue to stop dispatching events.
-   * Must only be called on the target thread, and with the innermost event
-   * queue.
-   */
-  [noscript] void popEventQueue(in nsIEventTarget aInnermostTarget);
 };
 
 /**